src/coredump/coredump.c

```diff
@@ -335,15 +376,11 @@ static int save_external_coredump(
         if (r < 0)
                 return log_error_errno(r, "Failed to determine coredump file name: %m");
 
-        r = tempfn_random(fn, NULL, &tmp);
-        if (r < 0)
-                return log_error_errno(r, "Failed to determine temporary file name: %m");
-
         mkdir_p_label("/var/lib/systemd/coredump", 0755);
```

Comment:
BTW:
(unrelated to this patch, just a reminder)
* there is no `mac_selinux_init`. So, `mkdir_p_label` is `selinux`-noop.
* `fix_permissions` doesn't call `selinux_label`